features/fixtures/mazerunner/app/src/main/java/com/bugsnag/android/mazerunner/MainActivity.kt

```diff
@@ -73,6 +76,33 @@ class MainActivity : Activity() {
         log("MainActivity.onResume complete")
     }
 
+    private fun setMazeRunnerAddress() {
+        val context = MazerunnerApp.applicationContext()
+        val externalFilesDir = context.getExternalFilesDir(null)
+        val configFile = File(externalFilesDir, "fixture_config.json")
+        log("Attempting to read Maze Runner address from config file ${configFile.path}")
+
+        // Poll for the fixture config file
+        val pollStart = System.currentTimeMillis()
+        while (System.currentTimeMillis() - pollStart < CONFIG_FILE_TIMEOUT) {
+            if (configFile.exists()) {
+                val fileContents = configFile.inputStream().use { it.reader().readText() }
+                val fixtureConfig = JSONObject(fileContents)
```

Comment:
While very unlikely, this could fail to parse the JSON file if it is still being written or is otherwise corrupt. In that case we would want to ignore the error and try again.

```suggestion
                val fixtureConfig = runCatching { JSONObject(fileContents) }.getOrNull()
```